pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/PinotDataType.java

```diff
@@ -507,9 +546,18 @@ public String toString(Object value) {
       Object[] valueArray = (Object[]) value;
       int length = valueArray.length;
       Integer[] integerArray = new Integer[length];
-      PinotDataType singleValueType = getSingleValueType();
-      for (int i = 0; i < length; i++) {
-        integerArray[i] = singleValueType.toInteger(valueArray[i]);
+      if (valueArray[0] instanceof Map) {
```

Comment:
The call will be something like this:

DataTypeTransformer will determine that source = PinotDataType.Map and dest = PinotDataType.STRING_ARRAY

We then call dest.convert(source), which enters the following:

```
STRING_ARRAY {
    @Override
    public String[] convert(Object value, PinotDataType sourceType) {
      return sourceType.toStringArray(value);
    }
  },
```

So now all we need is toStringArray, toIntegerArray etc on source. So instead of changing these functions for everyone, let's just override them for the new type you have introduced - INT_VALUE_MAP, LONG_VALUE_MAP etc.

Reply:
This can be another way, but it'll increase the number of overridden methods for EACH of the new PinotDataTypes. 
E.g. if the source type is STRING_VALUE_MAP and the destination type is INTEGER_ARRAY, the source data type has to have an overridden method of `toIntegerArray()`. I have a unit test on that, which is `DataTypeTransformerTest`.